src/convertToObject.js

```diff
@@ -12,7 +12,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let arr = sourceString.split(';');
+
+  arr = arr.map(property => property.trim());
+  arr = arr.filter(property => property.length > 0);
+
+  arr = arr.map(function makeArrays(property) {
+    return property.split(':');
+  });
```

Comment:
```suggestion
  let arr = sourceString.split(';')
    .map(property => property.trim())
    .filter(property => property.length > 0)
    .map(property => property.split(':'));
```

don't write functions inside methods. They already take callbacks that you can use